plans/upgrade.pp

```diff
@@ -56,42 +54,19 @@
     Target.new($target.merge($target_config)).add_to_group('peadm_nodes')
   }}
 
-  case $architecture {
-    'xlarge': {
-      $params = {
-        'master_host'                    => $inventory['master'][0]['name'],
-        'master_replica_host'            => $inventory['master'][1]['name'],
-        'puppetdb_database_host'         => $inventory['psql'][0]['name'],
-        'puppetdb_database_replica_host' => $inventory['psql'][1]['name'],
-        'compiler_hosts'                 => $inventory['compiler'].map |$c| { $c['name'] },
-        'console_password'               => $console_password,
-        'dns_alt_names'                  => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address'          => $apply['pool']['value'],
-        'version'                        => $version
-      }
-    }
-    'large': {
-      $params = {
-        'master_host'                    => $inventory['master'][0]['name'],
-        'compiler_hosts'                 => $inventory['compiler'].map |$c| { $c['name'] },
-        'console_password'               => $console_password,
-        'dns_alt_names'                  => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address'          => $apply['pool']['value'],
-        'version'                        => $version
-      }
-    }
-    'standard': {
-      $params = {
-        'master_host'                    => $inventory['master'][0]['name'],
-        'console_password'               => $console_password,
-        'dns_alt_names'                  => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address'          => $apply['pool']['value'],
-        'version'                        => $version
-      }
-    }
-    default: { fail('Something went horribly wrong or only xlarge is supported in this configuration') }
+  $params = {
```

Comment:
Is this duplicating code from init.pp? In that case I would suggest creating a function autope::peadm_params_from_inventory which accepts an inventory and returns params.